### PR TITLE
Remove `scala2-library-tasty-tests` project

### PR DESCRIPTION
**Pull request by nicolasstucki**
Split BootstrappedStdLibTASYyTest into the part that tests the TASTy inspector and the part that tests recompilation.

The `tests/run/scala2-library-test` test will be tested with the original Scala 2 library JAR and with the Scala 2 library TASTy JAR depending on the `scala2Library` SBT setting.

We also remove `scala2-library-tasty-tests` as it is now empty and will not serve any purpose anymore.

This test also discovered a bug in TASTYRun with the path separators of `JarArchive`s in Windows. The "/" is used in the JAR paths but the "\" is used for OS paths. This is now fixed.

[test_java8]

### CODE REVIEW

**Comment by nicolasstucki** [comment]
I had to make sure that the output of the _from tasty_ tests are set explicitly with `-d`. Otherwise, it generated the classes in the root directory.

File: tests/run-with-compiler/scala2-library-from-tasty.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    .map(_.normalize.path.stripPrefix(scalaLibClassesPath.toString + "/"))
+    .toList
+
+def compileFromTasty(blacklisted: Set[String]): Unit = {
+  val driver = new dotty.tools.dotc.Driver
+  val tastyFiles = scalaLibTastyPaths.filterNot(blacklisted)
+  val args = Array(
+    "-classpath", ClasspathFromClassloader(getClass.getClassLoader),
+    "-from-tasty",
+    "-d", "out/scala2-library-from-tasty-test-output.jar",
```
Comment (nicolasstucki) [inline tests/run-with-compiler/scala2-library-from-tasty.scala:39]: I added this line to this test.

File: tests/run-with-compiler/scala2-library-from-tasty-jar.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    .toList
+
+def compileFromTastyInJar(blacklisted: Set[String]): Unit = {
+  val driver = new dotty.tools.dotc.Driver
+  val yFromTastyBlacklist =
+    blacklisted.mkString("-Yfrom-tasty-ignore-list:", ",", "")
+  val args = Array(
+    "-classpath", ClasspathFromClassloader(getClass.getClassLoader),
+    "-from-tasty",
+    "-d", "out/scala2-library-from-tasty-jar-test-output.jar",
```
Comment (nicolasstucki) [inline tests/run-with-compiler/scala2-library-from-tasty-jar.scala:43]: I added this line to this test.

File: tests/run-with-compiler/scala2-library-from-tasty-jar.scala [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import dotty.tools.dotc.util.ClasspathFromClassloader
+
+import java.io.File.pathSeparator
+import java.io.File.separator
+
+@main def Test: Unit =
+  blacklistsOnlyContainsClassesThatExist()
+  // FIXME this test does not work on JDK8
+  // Caused by: dotty.tools.dotc.core.TypeError$$anon$1: package scala.quoted.runtime.Expr does not have a member method quote
+  if System.getProperty("java.specification.version") != "1.8" then
```
Comment (nicolasstucki) [inline tests/run-with-compiler/scala2-library-from-tasty-jar.scala:11]: This is a possible bug when running the from-TASTy compiler on JDK8 or with the test itself. I will open an issue to investigate this further.

Note that the Windows tests found a bug in `TASTYRun.scala` that was fixed but then failed due it running on JDK8. Therefore this fix will not be tested until we fix this issue or run Windows tests on a newer version of the JDK.

File: compiler/src/dotty/tools/dotc/fromtasty/TASTYRun.scala [hunk 4]
```diff
@@ -22,9 +22,9 @@ class TASTYRun(comp: Compiler, ictx: Context) extends Run(comp, ictx) {
       file.extension match
         case "jar" =>
           JarArchive.open(Path(file.path), create = false).allFileNames()
-            .map(_.stripPrefix(File.separator)) // change paths from absolute to relative
-            .filter(e => Path.extension(e) == "tasty" && !fromTastyIgnoreList(e))
-            .map(e => e.stripSuffix(".tasty").replace(File.separator, "."))
+            .map(_.stripPrefix("/")) // change paths from absolute to relative
+            .filter(e => Path.extension(e) == "tasty" && !fromTastyIgnoreList(e.replace("/", File.separator)))
```
Comment (bishabosha) [inline compiler/src/dotty/tools/dotc/fromtasty/TASTYRun.scala:26]: seems ok, but perhaps `fromTastyIgnoreList` should accept binary names (with `.` separator) rather than paths?

**Review by bishabosha — APPROVED** [review]

File: compiler/src/dotty/tools/dotc/fromtasty/TASTYRun.scala (hunk 4 above)
Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/fromtasty/TASTYRun.scala:26]: Maybe. Out of scope for this PR.
